### PR TITLE
Fix all of the eslint errors and warnings in packages/*

### PR DESCRIPTION
**Pull request by blowery**
#### Changes proposed in this Pull Request

* Fix the various eslint warnings and errors in packages/*

#### Testing instructions

* TBD

Fixes #

### CODE REVIEW

File: packages/eslint-plugin-wpcalypso/test-utils/format-message.js [hunk 1]
```diff
@@ -17,7 +17,7 @@
  */
 function formatMessage( message, terms ) {
 	return message.replace( /\{\{\s*(.+?)\s*\}\}/g, function( fullMatch, term ) {
-		if ( terms.hasOwnProperty( term ) ) {
+		if ( Object.prototype.hasOwnProperty.call( terms, term ) ) {
```
[the remark this answers is not in the document]
Comment (sirbrillig) [reply packages/eslint-plugin-wpcalypso/test-utils/format-message.js]: Oh, nevermind. I just saw the link in the commit message 🤦‍♂

[the remark this answers is not in the document]
Comment (blowery) [reply packages/eslint-plugin-wpcalypso/test-utils/format-message.js]: I'm not in love with this rule. It's a fairly new one that comes with `eslint:recommended`...

File: packages/eslint-plugin-wpcalypso/lib/util/test/get-i18n-string-from-node.js [hunk 2]
```diff
@@ -20,60 +19,53 @@ function parseExpressionStatement( code ) {
 	return node;
 }
 
-describe( '#getStringFromNode', function() {
-	it( 'should return simple strings', function() {
-		assert.equal(
-			'a simple string',
-			getTextContentFromNode( parseExpressionStatement( "'a simple string'" ) )
+describe( '#getStringFromNode', () => {
+	test( 'should return simple strings', () => {
```
[the remark this answers is not in the document]
Comment (blowery) [reply packages/eslint-plugin-wpcalypso/lib/util/test/get-i18n-string-from-node.js]: That's just what jest prefers. We hacked in `it` support when we migrated from mocha and the jest eslint linter doesn't understand it.

[the remark this answers is not in the document]
Comment (sirbrillig) [reply packages/eslint-plugin-wpcalypso/lib/util/test/get-i18n-string-from-node.js]: Ahhh! TIL.

[the remark this answers is not in the document]
Comment (blowery) [reply packages/eslint-plugin-wpcalypso/lib/util/test/get-i18n-string-from-node.js]: hmmm maybe i'm not remembering correctly. jest does appear to alias `it` to `test`, though they don't really mention `it` anywhere else in the docs. https://jestjs.io/docs/en/api#testname-fn-timeout

[the remark this answers is not in the document]
Comment (blowery) [reply packages/eslint-plugin-wpcalypso/lib/util/test/get-i18n-string-from-node.js]: I think this one was actually moved by the jestcodmod I initially ran against this set of tests...

File: packages/eslint-plugin-wpcalypso/test-utils/format-message.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsnajdr) [reply packages/eslint-plugin-wpcalypso/test-utils/format-message.js]: Being a friend of Lodash, I can't help recommending `has( terms, term )`, which uses `hasOwnProperty.call` internally :wink:

[the remark this answers is not in the document]
Comment (blowery) [reply packages/eslint-plugin-wpcalypso/test-utils/format-message.js]: https://github.com/Automattic/wp-calypso/pull/37687 to nuke the rule 💥

**Comment by lancewillett** [comment]
Cleaning up older pull requests as part of a GitHub scrub.

Is this change still valid? What's needed to move this along?

If not valid, please close with a comment. Thanks!

**Comment by scinos** [comment]
Outdated

**Pull request closed without merging**